### ISSUE TITLE
Byte.parseHex(String str) could throw IllegalArgumentException

### ISSUE THREAD

**Issue by petrukhnov** [patrickfav/bytes-java#37]
Byte.parseHex(String str) could throw IllegalArgumentException on malformed string. It is not mentioned in docs or javadoc. 

```
java.lang.IllegalArgumentException: invalid hex string, must be mod 2 == 0
	at at.favre.lib.bytes.BinaryToTextEncoding$Hex.decode(BinaryToTextEncoding.java:108)
	at at.favre.lib.bytes.Bytes.parse(Bytes.java:652)
	at at.favre.lib.bytes.Bytes.parseHex(Bytes.java:603)


```

**Comment by patrickfav** [comment]
Thanks for reporting, I will add a hint in the javadoc

**Comment by patrickfav** [comment]
On a different thought, since the decoder is very forgiving, I added the feature that it also accepts odd number length strings with auto padding of zero, so currently it supports:

* Upper- and lowercase `a-f` (also mixed case)
* Prefix with `0x` which will be ignored
* Even and odd number of string length with auto zero padding (ie. 'E3F' is same as '0E3F')

**Comment by patrickfav** [comment]
Will be released with 1.2.0
